applications/callflow/src/module/cf_route_to_cid.erl

```diff
@@ -0,0 +1,72 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2011-2017, 2600Hz INC
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%   Kirill Sysoev
+%%%-------------------------------------------------------------------
+-module(cf_route_to_cid).
+
+-behaviour(gen_cf_action).
+
+-include("callflow.hrl").
+
+-export([handle/2]).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% This module attempts to lookup endpoints by it's cid number.
+%% Returns continue if fails to connect or stop when successful.
+%% @end
+%%--------------------------------------------------------------------
+-spec handle(kz_json:object(), kapps_call:call()) -> 'ok'.
+handle(Data, Call) ->
+    case endpoints_lookup(Data, Call) of
+        [] ->
+            lager:debug("~p not found", [kapps_call:callee_id_number(Call)]),
+            cf_exe:continue(Call);
+        EndpointJObjs ->
+            Vals = [{<<"strategy">>,<<"simultaneous">>}
+                   ,{<<"endpoints">>, EndpointJObjs}
+                   ,{<<"timeout">>,kz_json:get_integer_value(<<"timeout">>, Data, ?DEFAULT_TIMEOUT_S)}
+                   ],
+            cf_group:handle(kz_json:set_values(Vals, Data), Call)
+    end.
+
+-spec endpoints_lookup(kz_json:object(), kapps_call:call()) -> kz_proplist().
+endpoints_lookup(Data, Call) ->
+    lists:foldl(fun(CIDType, Acc) -> cid_type_based_lookup(CIDType, Data, Call) ++ Acc end
+               ,[]
+               ,kz_json:get_value(<<"cid_types">>, Data, [])
```

Comment:
Do you want this to create 0 endpoints if no "cid_types" restriction exists in `Data`? Perhaps it should default to `[<<"device">>]` or `[<<"device">>, <<"user">>]`?

Reply:
Sorry, should be `["internal"]` or `["internal", "external"]` - not sure what happened :)